django/db/backends/mysql/creation.py

```diff
@@ -11,3 +16,39 @@ def sql_table_creation_suffix(self):
         if test_settings['COLLATION']:
             suffix.append('COLLATE %s' % test_settings['COLLATION'])
         return ' '.join(suffix)
+
+    def _clone_test_db(self, number, verbosity, keepdb=False):
+        qn = self.connection.ops.quote_name
+        source_database_name = self.connection.settings_dict['NAME']
+        target_database_name = self.get_test_db_clone_settings(number)['NAME']
+
+        with self._nodb_connection.cursor() as cursor:
+            try:
+                cursor.execute(
+                    "CREATE DATABASE %s" % qn(target_database_name))
+            except Exception as e:
+                if keepdb:
+                    return
+                try:
+                    if verbosity >= 1:
+                        print("Destroying old test database '%s'..."
+                              % self.connection.alias)
+                    cursor.execute(
+                        "DROP DATABASE %s" % qn(target_database_name))
+                    cursor.execute(
+                        "CREATE DATABASE %s" % qn(target_database_name))
+                except Exception as e:
+                    sys.stderr.write(
+                        "Got an error recreating the test database: %s\n" % e)
+                    sys.exit(2)
+
+        dump_cmd = DatabaseClient.settings_to_cmd_args(self.connection.settings_dict)
```

Comment:
I added a commit that allows you to inject a custom `ParallelTestSuite` class and to customize both how workers are initialized and how they run tests.

(That said, I still believe that you can achieve your goals just by creating the database outside of the test runner and using `--keepdb`.)